src/Compilers/CSharp/Portable/Symbols/Source/ImplicitNamedTypeSymbol.cs

```diff
@@ -71,6 +76,12 @@ protected override void CheckBase(DiagnosticBag diagnostics)
             }
         }
 
+        protected override void CheckExtensionClass(DiagnosticBag diagnostics)
+        {
+            var extensionClass = this.ExtensionClassTypeNoUseSiteDiagnostics;
```

Comment:
Inline `ExtensionClassTypeNoUseSiteDiagnostics` in `Debug.Assert` to avoid making the call in release builds.